packages/near-membrane-base/src/environment.ts

```diff
@@ -1,192 +1,181 @@
-                    // first call we don't pay the cost of wrapping anymore?
-                    //
-                    // TODO: Isn't it easier to just not do any lazy stuff anymore considering
-                    // that the creation of those proxies is now faster?
-                    const blueDistortedGetter = this.distortionCallback(
-                        blueGetter
-                    ) as () => BlueValue;
-                    currentBlueGetter = function currentDistortedBlueGetter() {
-                        const value: BlueValue = ReflectApply(
-                            blueDistortedGetter,
-                            broker.getBlueValue(this),
-                            emptyArray
-                        );
-                        return broker.getRedValue(value);
-                    };
-                    redDescriptor.get = function get(): RedValue {
-                        return ReflectApply(currentBlueGetter, this, emptyArray);
-                    };
-                }
+            const configurable = ObjectHasOwnProperty.call(blueDescriptor, 'configurable')
+                ? !!blueDescriptor.configurable
+                : undefinedSymbol;
+            const enumerable = ObjectHasOwnProperty.call(blueDescriptor, 'enumerable')
+                ? !!blueDescriptor.enumerable
+                : undefinedSymbol;
+            const writable = ObjectHasOwnProperty.call(blueDescriptor, 'writable')
+                ? !!blueDescriptor.writable
+                : undefinedSymbol;
+            const valuePointer = ObjectHasOwnProperty.call(blueDescriptor, 'value')
+                ? this.blueGetTransferableValue(blueDescriptor.value)
+                : undefinedSymbol;
+            const getPointer = ObjectHasOwnProperty.call(blueDescriptor, 'get')
+                ? this.blueGetTransferableValue(blueDescriptor.get)
+                : undefinedSymbol;
+            const setPointer = ObjectHasOwnProperty.call(blueDescriptor, 'set')
+                ? this.blueGetTransferableValue(blueDescriptor.set)
+                : undefinedSymbol;
+            // installing descriptor into the red side
+            this.redCallableDefineProperty(
+                oPointer,
+                key,
+                configurable,
+                enumerable,
+                writable,
+                valuePointer,
+                getPointer,
+                setPointer
+            );
+        }
+    }
 
-                if (typeof blueDescriptor.set === 'function') {
-                    redDescriptor.set = function set(v: RedValue): void {
-                        // if a global setter is invoke, the value will be use as it
-                        // is as the result of the getter operation
-                        currentBlueGetter = () => v;
-                    };
+    lazyRemap(o: ProxyTarget, keys: PropertyKey[]) {
+        const enumerablePropertyKeys = Object.keys(o); // except symbols
```

Comment:
☝️ use a reference to `Object.keys` and not `Object.keys` directly.

Reply:
Actually, we can just remove `lazyRemap` altogether since it's not being used

Reply:
oh no, lazyRemap is part of the optimizations we need to do. it is not used yet, but it will be used to match the existing implementation.